Imagine/Filter/Loader/DownscaleFilterLoader.php

```diff
@@ -30,7 +30,16 @@ public function load(ImageInterface $image, array $options = array())
             $widthRatio = $width / $origWidth;
             $heightRatio = $height / $origHeight;
 
-            $ratio = min($widthRatio, $heightRatio);
+            // faster check than is_null
+            if ($width === null || $height === null) {
+                $ratio = max($widthRatio, $heightRatio);
+            } else {
+                $ratio = min($widthRatio, $heightRatio);
+            }
+
+            if ($ratio > 1) {
+                return $image;
+            }
```

Comment:
I was thinking more like, creating another filter "scale" which will do either, the filters should do only what they should, I don't know maybe it's just clear that way.